adam_core/utils/helpers/data/get_test_data.py

```diff
@@ -141,3 +147,50 @@ def _get_orbital_elements(
                     ),
                     index=False,
                 )
+
+    # Lets query for propagated Horizons state vectors
+    propagated_orbits_list = []
+    ephemeris_dfs = []
+
+    # Create an array of delta times to propagate the orbits and
+    # get ephemerides relative to the epoch at which the orbits are defined
+    total_days = 60
+    half_arc = total_days / 2
+    # Make it so there are 3 observations every 2 days
+    dts = np.arange(-half_arc, half_arc, 2)
+    dts = np.concatenate([dts, dts + 1 / 48, dts + 1 / 24])
+    dts.sort()
+    num_dts = len(dts)
+
+    for i, (object_id, orbit) in enumerate(zip(object_ids, orbits)):
+        # Extract times from orbit and propagate +/- 30 days
+        times = orbit.coordinates.time.to_astropy() + dts * u.day
+
+        # Query for propagated Horizons state vectors
+        propagated_orbit = query_horizons([object_id], times)
+        propagated_orbit = propagated_orbit.set_column(
+            "orbit_id", pa.array([f"{i:05d}" for _ in range(len(times))])
+        )
+
+        # Define two observers one at the Rubin Observatory and one at CTIO
+        observer_X05 = Observers.from_code("X05", times[: num_dts // 2])
+        observer_W84 = Observers.from_code("W84", times[num_dts // 2 :])
+        observers = qv.concatenate([observer_X05, observer_W84])
+
+        # Query for ephemeris at the same times
+        ephemeris = query_horizons_ephemeris([object_id], observers)
+        ephemeris["orbit_id"] = [f"{i:05d}" for _ in range(len(times))]
+
+        propagated_orbits_list.append(propagated_orbit)
+        ephemeris_dfs.append(ephemeris)
+
+    propagated_orbits: Orbits = qv.concatenate(propagated_orbits_list)
+    propagated_orbits.to_dataframe().to_csv(
+        files("adam_core.utils.helpers.data").joinpath("propagated_orbits.csv"),
+        index=False,
+    )
```

Comment:
Why the intermediate to_dataframe, instead of straight to CSV?

Reply:
The main reason is I didn't know `qv.Table` had a `.to_csv`. But also, the adam_core orbits class has its own definition of `.to_dataframe` that flattens the covariance matrix into a lower triangular representation with appropriate column names. Not sure how much we care about the legibility of the testing files though.

Reply:
It looks like I pulled some shenanigans in the Times class as well that [encodes the time scale as part of the column names](https://github.com/B612-Asteroid-Institute/adam_core/blob/main/adam_core/coordinates/times.py#L61). I think we might want to revisit some of the decisions I've made with regard to `.to_dataframe` and `.from_dataframe` in some of the adam_core classes. I think it's likely that `to_csv` using quivr is easy and legible enough that we could go through and remove the dataframe implementations here.